.circleci/config.yml

```diff
@@ -0,0 +1,127 @@
+          "$CIRCLE_BUILD_URL"
+          > version.json
+      - run: docker build --pull -t addons-frontend .
+      - run: docker images
+      - run:
+          name: Write the sha256 sum to an artifact for verification
+          command: |
+            docker images --no-trunc | awk '/^app/ {print $3}' | sudo tee $CIRCLE_ARTIFACTS/docker-image-shasum256.txt
+      - run:
+          name: Save new image
+          command: |
+            IMG="image-$(date +%j).gz"; docker save addons-frontend | pigz --fast -c > ~/addons-frontend/docker/$IMG; ls -l ~/addons-frontend/docker
+            ls -l ~/addons-frontend/docker
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./*
+  integration-test:
+    working_directory: ~/addons-frontend
+    machine: true
+    steps:
+      - checkout
+      - run:
+          name: Install Docker Compose
+          command: |
+            set -x
+            pip install docker-compose>=1.18
+            docker-compose --version
+      - run:
+          name: Run tests
+          command: ./tests/ui/run-functional-tests.sh
+      - store_artifacts:
+          path: addons-server/ui-test.html
+  release-tag: # build for releases (tags)
+    machine: true
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Load saved image
+          command: docker load -i docker/image-*
+      - run:
+          name: Build docker image and push to repo
+          command: |
+            [ ! -z $DOCKERHUB_REPO ]
+            docker login -u $DOCKER_USER -p $DOCKER_PASS
+            echo $DOCKERHUB_REPO:$CIRCLE_TAG
+            docker tag addons-frontend $DOCKERHUB_REPO:$CIRCLE_TAG
+            docker images
+            docker push $DOCKERHUB_REPO:$CIRCLE_TAG
+  release-master: # build for master
+    machine: true
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: Load saved image
+          command: docker load -i docker/image-*
+      - run:
+          name: Build docker image and push to repo
+          command: |
+            [ ! -z $DOCKERHUB_REPO ]
+            docker login -u $DOCKER_USER -p $DOCKER_PASS
+            docker tag addons-frontend $DOCKERHUB_REPO:latest
+            docker push $DOCKERHUB_REPO:latest
+workflows:
+  version: 2
+  build_deploy_release:
+    jobs:
+      - build:
+          filters:
+            branches:
+              only: master
+            tags:
+              only: /.*/
+      - release-tag:
+          requires:
+              - build
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              ignore: /.*/
+      - release-master:
+          requires:
+            - build
+          filters:
+            branches:
+              only: master
+  scheduled-workflow:
+    triggers:
+      - schedule:
+          cron: "0 0,2 * * *"
+          filters:
+            branches:
+              only:
+                - master
```

Comment:
I do think we want that for add-ons server too :heart_eyes: but both works for me really, it's just that the tests tend to be failing more than needed with intermittent failures and people tend to not look at them anymore.